### PR TITLE
Behavior

### PR DESCRIPTION
**Pull request by delamux**
create behavior with review of Rodrigo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,3 +9,37 @@ The recommended way to install composer packages is:
...
+* Use the behavior in your afterSave function
+
+```
+public function afterSave(Event $event, EntityInterface $entity, $options)
+    {
+        return $this->addMessage($entity->id, [
+            'subject' => 'This is the subject',
+            'body' => 'This is the body' .
+                $entity->first_name . ' ' . $entity->last_name .
+                ' the message has been sended to ' . $entity->email
```
Comment (steinkel) [inline README.md]: should be 
' the message has been sent to ' . $entity->email

File: src/Model/Behavior/MessagesBehavior.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+     * -model //you can to assign table name or get automatically from model
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'subject' => 'Message subject',
+        'body' => 'This is the body'
+    ];
+
+    public function addMessage($id, array $data){
+        $config = $this->_defaultConfig;
```
Comment (steinkel) [inline src/Model/Behavior/MessagesBehavior.php]: no need to do this, InstanceConfigTrait and Behavior should have done that before

File: src/Model/Behavior/MessagesBehavior.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+     * -body
+     * -model //you can to assign table name or get automatically from model
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'subject' => 'Message subject',
+        'body' => 'This is the body'
+    ];
+
+    public function addMessage($id, array $data){
```
Comment (steinkel) [inline src/Model/Behavior/MessagesBehavior.php]: missing docblock
var $id, I would change this to $foreignKey instead
var $data, would be better to allow passing entities too

File: src/Model/Behavior/MessagesBehavior.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        $config = $this->_defaultConfig;
+        Hash::merge($config, $data);
+
+        $messagesTable = TableRegistry::get('CakeMojo/Messages.Messages');
+
+        $message = $messagesTable->newEntity();
+
+        $message->foreign_key = $id;
+        $message->subject = $data['subject'];
```
Comment (steinkel) [inline src/Model/Behavior/MessagesBehavior.php]: what if there is no 'subject' key, we need to check it

File: src/Model/Behavior/MessagesBehavior.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        $message->body = $data['body'];
+        $message->model = empty($data['model']) ? $data['model'] = $this->_table->getTable() : $data['model'];
```
Comment (steinkel) [inline src/Model/Behavior/MessagesBehavior.php]: ternary is hard to read and understand, I usually remove complex ones like here

File: src/Model/Behavior/MessagesBehavior.php [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+        return $messagesTable->save($message);
+
```
Comment (steinkel) [inline src/Model/Behavior/MessagesBehavior.php]: phpcs issues

**Review by steinkel — CHANGES_REQUESTED** [review]

File: src/Model/Behavior/MessagesBehavior.php (hunk 4 above)
Comment (delamux) [reply src/Model/Behavior/MessagesBehavior.php]: Ok, in check function for this parameter if not exist, I show Exception error or subject string by default, is not necessary check if I pass a default configuration.

Comment (delamux) [reply src/Model/Behavior/MessagesBehavior.php]: @chav170 resolved the problem (Magic Hash::get)

**Pull request merged**